### ISSUE TITLE
Incremental history index duplicates records when OpenGroklatestRev does not exist

### ISSUE THREAD

**Issue by tulinkry** [oracle/opengrok#2734]
When file `OpenGroklatestRev` does not exist, yet the history cache is created, the incremental index will duplicate all the records in the history cache.

Expected behaviour would be that the existing history cache is overwritten.

**Comment by idodeclare** [comment]
I am working on a refactoring of `hist` so that OpenGrok doesn't have to load a repository's entire history into memory at once. My use case is speeding up the super-slow, full reindexing of freebsd, which has a long history and substantial number of renamings. (Improving handling of renamings as well is covered, as currently OpenGrok does for all files a linear search through the history of renamings; for freebsd that history is hundreds of thousands of files.)

Anyway the problem of this issue will be handled implicitly by my patch.

I hadn't thought to try any de-duplicating when for example `OpenGroklatestRev` exists but fails to update after an incremental reindexing — but we could think about possibly doing that.

**Comment by Leo-Adlakha** [comment]
Hi, I am experiencing similar behavior in my instance, any updates on this ?

**Comment by vladak** [comment]
> Hi, I am experiencing similar behavior in my instance, any updates on this ?

Can you check if the `historycache` directory (for given project) contains the `OpenGroklatestRev` file ?

**Comment by Leo-Adlakha** [comment]
> Can you check if the `historycache` directory (for given project) contains the `OpenGroklatestRev` file ?

Yes, it contains `OpenGroklatestRev` file. One more thing to add the duplication seems to be proportional to number of commits being added to the file.

**Comment by vladak** [comment]
> > Can you check if the `historycache` directory (for given project) contains the `OpenGroklatestRev` file ?
> 
> Yes, it contains `OpenGroklatestRev` file. One more thing to add the duplication seems to be proportional to number of commits being added to the file.

Then it's different problem. This is specifically tracking the case when the `OpenGroklatestRev` file is missing.

**Comment by msillence** [comment]
I think this might be resolved in 1.8.2
We just experienced the same issue and upgrading to 1.8.2 it seems to have gone

**Comment by dpsi** [comment]
Do we need to have history enabled to be affected by this bug?
I am seeing duplicate search results on 1.9.2, but I'm not sure if this is the issue that I should be monitoring since I don't have history enabled.
